src/amp-story-embed.js

```diff
@@ -88,33 +125,67 @@ export class AmpStoryEmbed {
   }
 
   /**
-   * @param {number} index
+   * @param {!Element} story
    * @private
    */
-  buildIframe_(index) {
-    const story = this.stories_[index];
-    this.iframeEl_ = this.doc_.createElement('iframe');
+  buildIframe_(story) {
+    const iframeEl = this.doc_.createElement('iframe');
     setStyle(
-      this.iframeEl_,
+      iframeEl,
       'backgroundImage',
       story.getAttribute('data-poster-portrait-src')
     );
-    this.iframeEl_.classList.add('story');
-    this.initializeLoadingListeners_();
-    this.rootEl_.appendChild(this.iframeEl_);
+    iframeEl.setAttribute(IFRAME_ID_ATTR_NAME, 'i' + this.iframeIdCounter_++);
+    iframeEl.classList.add('story-embed-iframe');
+
+    this.initializeLoadingListeners_(iframeEl);
+    this.rootEl_.appendChild(iframeEl);
+
+    this.initializeHandshake_(story, iframeEl).then(
+      messaging => {
+        const iframeId = iframeEl[IFRAME_ID_ATTR_NAME];
+        this.messagingFor_[iframeId] = messaging;
+
+        // TODO(Enriqe): Appropiately set visibility to stories.
+        this.displayStory_(iframeId);
+      },
+      err => {
+        console /*OK*/
+          .log({err});
+      }
+    );
+  }
+
+  /**
+   * @param {!Element} story
+   * @param {!Element} iframeEl
+   * @return {Promise<Messaging>}
+   * @private
+   */
+  initializeHandshake_(story, iframeEl) {
+    const frameOrigin = this.getEncodedUrl_(story.href).origin;
+
+    return Messaging.waitForHandshakeFromDocument(
+      this.win_,
+      iframeEl.contentWindow,
+      frameOrigin
+    );
   }
 
-  /** @private  */
-  initializeLoadingListeners_() {
+  /**
+   * @param {!Element} iframeEl
+   * @private
+   */
+  initializeLoadingListeners_(iframeEl) {
     this.rootEl_.classList.add(LoadStateClass.LOADING);
 
-    this.iframeEl_.onload = () => {
+    iframeEl.onload = () => {
       this.rootEl_.classList.remove(LoadStateClass.LOADING);
       this.element_.classList.remove(LoadStateClass.LOADING);
```

Comment:
Not your doing, but this class doesn't seem to be ever added. Can we remove these statements?